### ISSUE TITLE
"export 'WebstorableArray' was not found in 'ngx-store'

### ISSUE THREAD

**Issue by RChoque** [zoomsphere/ngx-store#15]
I need to save array of objects, so WebstorableArray seems to be perfects for that, but I have compilation warnings : 

WARNING in ./src/app/order/order.component.ts
656:55-71 "export 'WebstorableArray' was not found in 'ngx-store'

WARNING in ./src/app/order/order.component.ts
656:91-107 "export 'WebstorableArray' was not found in 'ngx-store'

WARNING in ./src/app/order/order.component.ts
660:55-71 "export 'WebstorableArray' was not found in 'ngx-store'

WARNING in ./src/app/order/order.component.ts
660:91-107 "export 'WebstorableArray' was not found in 'ngx-store'

WARNING in ./src/app/order/order.component.ts
668:55-71 "export 'WebstorableArray' was not found in 'ngx-store'

WARNING in ./src/app/order/order.component.ts
668:91-107 "export 'WebstorableArray' was not found in 'ngx-store'

**Comment by DanielKucal** [comment]
Hi @RChoque, thanks for your interest in `ngx-store`. Do you have latest version installed?
Also please note that `WebstorableArray<T>` is a generic type that takes an argument just like regular `Array<T>` type.
```
export type WebstorableArray<T> = Webstorable & Array<T>;
```

**Comment by tuur29** [comment]
I get the exact same error with the latest version:

```
WARNING in ./src/app/order/order.component.ts
92:50-66 "export 'WebstorableArray' was not found in 'ngx-store'
    at HarmonyImportSpecifierDependency._getErrors (C:\github\repo\node_modules\webpack\lib\dependencies\HarmonyImportSpecifierDependency.js:65:15)
    at HarmonyImportSpecifierDependency.getWarnings (C:\github\repo\node_modules\webpack\lib\dependencies\HarmonyImportSpecifierDependency.js:39:15)
    at Compilation.reportDependencyErrorsAndWarnings (C:\github\repo\node_modules\webpack\lib\Compilation.js:694:24)
    at Compilation.finish (C:\github\repo\node_modules\webpack\lib\Compilation.js:552:9)
    at applyPluginsParallel.err (C:\github\repo\node_modules\webpack\lib\Compiler.js:512:17)
    at C:\github\repo\node_modules\tapable\lib\Tapable.js:289:11
    at _addModuleChain (C:\github\repo\node_modules\webpack\lib\Compilation.js:498:11)
    at processModuleDependencies.err (C:\github\repo\node_modules\webpack\lib\Compilation.js:468:14)
    at _combinedTickCallback (internal/process/next_tick.js:95:7)
    at process._tickCallback (internal/process/next_tick.js:161:9)
```

My code in order.component.ts:

```
import { LocalStorage, WebstorableArray } from 'ngx-store';
...

@LocalStorage() order: WebstorableArray<any> = <any>[];
...

resetOrder() {  this.order = <any>[]; }
editOrder() {
  ...

  this.order.save();
}
```

**Comment by DanielKucal** [comment]
That's really weird... I've just checked it with JiT and AoT compiler and everything works just fine. 
@tuur29, can you please paste here content of file `node_modules/ngx-store/dist/src/index.d.ts` from your project? Second question, do you use angular-cli or some custom webpack environment?

**Comment by tuur29** [comment]
I use angular-cli.

index.d.ts:
```
export { CookieStorage, LocalStorage, SessionStorage, SharedStorage } from './decorator/webstorage';
export { WebStorageService, CookiesStorageService, LocalStorageService, SessionStorageService, SharedStorageService } from './service';
export { WebStorageConfigInterface, WEBSTORAGE_CONFIG } from './config';
export declare class Webstorable {
    save(): void;
}
export declare type WebstorableObject = Webstorable & Object;
export declare type WebstorableArray<T> = Webstorable & Array<T>;
export declare class WebStorageModule {
}
```

**Comment by gustavooferreira** [comment]
Hi,

I'm getting the same warning:

```
WARNING in ./src/app/services/devices-view-history.service.ts
42:50-66 "export 'WebstorableArray' was not found in 'ngx-store'
    at HarmonyImportSpecifierDependency._getErrors (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:65:15)
    at HarmonyImportSpecifierDependency.getWarnings (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:39:15)
    at Compilation.reportDependencyErrorsAndWarnings (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/Compilation.js:694:24)
    at Compilation.finish (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/Compilation.js:552:9)
    at applyPluginsParallel.err (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/Compiler.js:512:17)
    at /home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/tapable/lib/Tapable.js:289:11
    at _addModuleChain (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/Compilation.js:498:11)
    at processModuleDependencies.err (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/Compilation.js:468:14)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)

WARNING in ./src/app/services/devices-view-history.service.ts
42:86-102 "export 'WebstorableArray' was not found in 'ngx-store'
    at HarmonyImportSpecifierDependency._getErrors (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:65:15)
    at HarmonyImportSpecifierDependency.getWarnings (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/dependencies/HarmonyImportSpecifierDependency.js:39:15)
    at Compilation.reportDependencyErrorsAndWarnings (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/Compilation.js:694:24)
    at Compilation.finish (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/Compilation.js:552:9)
    at applyPluginsParallel.err (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/Compiler.js:512:17)
    at /home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/tapable/lib/Tapable.js:289:11
    at _addModuleChain (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/Compilation.js:498:11)
    at processModuleDependencies.err (/home/gustavoferreira/WebstormProjects/microACS-gui/node_modules/webpack/lib/Compilation.js:468:14)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

I'm creating an empty array, that's all I'm doing:

```
@LocalStorage() private devices: WebstorableArray<string> = <any>[];
```

I'm also using the latest version of Angular and ngx-store.

**Comment by DanielKucal** [comment]
I've found this issue on Angular's repo: https://github.com/angular/angular-cli/issues/2034
td;dr: wait for Angular 5...

**Comment by DanielKucal** [comment]
I've found this issue on Angular's repo: https://github.com/angular/angular-cli/issues/2034
tl;dr: wait for Angular 5...

**Comment by DanielKucal** [comment]
BTW is it still a problem with v1.4+?
